cpp/ql/src/experimental/Security/CWE/CWE-297/ValidationCertificateHostMismatch.ql

```diff
@@ -0,0 +1,141 @@
+
+FunctionCall getParentCall(FunctionCall f1, FunctionCall f2) {
+  result = f1 and f1.getEnclosingFunction() = f2.getEnclosingFunction()
+  or
+  result = getParentCall(f1.getEnclosingFunction().getACallToThisFunction(), f2)
+  or
+  result = getParentCall(f1, f2.getEnclosingFunction().getACallToThisFunction())
+}
+
+predicate checkHostnameOlder1(FunctionCall fc) {
+  exists(FunctionCall fctmp1, FunctionCall fcNode |
+    fcNode = getParentCall(fctmp1, fc) and
+    fctmp1.getTarget().hasName("X509_get_ext_d2i") and
+    (
+      exists(MacroInvocation mi1tmp, MacroInvocation mi2tmp |
+        mi1tmp.getMacroName() = "sk_GENERAL_NAME_num" and
+        mi2tmp.getMacroName() = "sk_GENERAL_NAME_value"
+      )
+      or
+      exists(FunctionCall fctmp2, FunctionCall fctmp3 |
+        fctmp2.getTarget().hasName("sk_GENERAL_NAME_num") and
+        fctmp3.getTarget().hasName("sk_GENERAL_NAME_value")
+      )
+    )
+  )
+}
+
+predicate checkHostnameOlder2(FunctionCall fc) {
+  exists(
+    FunctionCall fctmp1, FunctionCall fctmp2, FunctionCall fctmp3, FunctionCall fctmp4,
+    FunctionCall fcNode
+  |
+    fcNode = getParentCall(fctmp1, fc) and
+    fctmp3 != fctmp4 and
+    globalValueNumber(fctmp1) = globalValueNumber(fctmp3.getArgument(0)) and
+    globalValueNumber(fctmp2) = globalValueNumber(fctmp4.getArgument(0)) and
+    fctmp1.getTarget().hasName("X509_get_subject_name") and
+    fctmp2.getTarget().hasName("X509_get_subject_name") and
+    fctmp3.getTarget().hasName("X509_NAME_get_text_by_NID") and
+    (
+      fctmp4.getTarget().hasName("X509_NAME_get_text_by_NID") or
+      fctmp4.getTarget().hasName("X509_NAME_get_entry")
+    )
+  )
+}
+
+predicate checkHostnameUp110(FunctionCall fc) {
+  exists(FunctionCall fctmp1, FunctionCall fctmp2, FunctionCall fctmp3, FunctionCall fcNode |
+    fcNode = getParentCall(fctmp1, fc) and
+    globalValueNumber(fctmp1.getArgument(0)) = globalValueNumber(fctmp2.getArgument(0)) and
+    globalValueNumber(fctmp2.getArgument(0)) = globalValueNumber(fctmp3.getArgument(0)) and
+    fctmp1.getTarget().hasName("SSL_set_hostflags") and
+    fctmp2.getTarget().hasName("SSL_set1_host") and
+    fctmp3.getTarget().hasName("SSL_set_verify")
+  )
+}
+
+predicate checkHostnameUp102(FunctionCall fc) {
+  exists(
+    FunctionCall fctmp1, FunctionCall fctmp2, FunctionCall fctmp3, FunctionCall fctmp4,
+    FunctionCall fcNode
+  |
+    fcNode = getParentCall(fctmp1, fc) and
+    globalValueNumber(fctmp1.getArgument(0)) = globalValueNumber(fctmp4.getArgument(0)) and
+    globalValueNumber(fctmp1) = globalValueNumber(fctmp2.getArgument(0)) and
+    globalValueNumber(fctmp1) = globalValueNumber(fctmp3.getArgument(0)) and
+    fctmp1.getTarget().hasName("SSL_get0_param") and
+    fctmp2.getTarget().hasName("X509_VERIFY_PARAM_set_hostflags") and
+    fctmp3.getTarget().hasName("X509_VERIFY_PARAM_set1_host") and
+    fctmp4.getTarget().hasName("SSL_set_verify")
+  )
+}
+
+predicate checkHostnameGnuTLS(FunctionCall fc) {
```

Comment:
No. The Code Scanning alert is saying that QL names must preferably be in PascalCase (for classes and modules) or camelCase (for predicates and variables). So `checkHostnameGnuTLS` should ideally be `checkHostnameGnuTls` to match our [style guide](https://github.com/github/codeql/blob/main/docs/ql-style-guide.md). However, one could argue that, since `TLS` is a common way to write Transport Layer Security, this name is actually fine.

So I don't think you need to change your name to make the Code Scanning alert disappear.